### PR TITLE
Adding Versa.finance

### PR DESCRIPTION
**Pull request by versa-finance-dev**
##### methodology (what is being counted as tvl, how is tvl being calculated):
User locked Versa or Versa LP to dapp contract and claim rewards

### CODE REVIEW

File: projects/Versa/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+const TOKENS = {
+    WASTAR: NATIVE_TOKEN_WASTAR,
+    VERSA: "0xB9dEDB74bd7b298aBf76b9dFbE5b62F0aB05a57b",
+    USDC: "0x6a2d262D56735DbA19Dd70682B39F6bE9a931D98",
+    BUSD: "0x4Bf769b05E832FCdc9053fFFBC78Ca889aCb5E1E",
+}
+
+const versaTvl = calculateUsdUniTvl(
+    FACTORIES,
+    "versa",
```
Comment (g1nt0ki) [inline projects/Versa/index.js:16]: can you change this to `astar`, thats the chain name :)

File: projects/Versa/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    NATIVE_TOKEN_WASTAR,
+    [
+        ...Object.values(TOKENS)
+    ], "versa")
```
Comment (g1nt0ki) [inline projects/Versa/index.js:20]: same here versa -> astar

File: projects/Versa/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+module.exports = {
+    timetravel: true,
+    methodology: "Versa Tvl Calculation",
+    tvl: versaTvl
```
Comment (g1nt0ki) [inline projects/Versa/index.js:25]: can you put the tvl under chain name, like

    methodlogy: ...
    astar: { tvl: versaTvl }

**Review by g1nt0ki — CHANGES_REQUESTED** [review]

**Comment by versa-finance-dev** [comment]
@g1nt0ki Thanks, updated, check please

File: projects/Versa/index.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    "astar",
+    NATIVE_TOKEN_WASTAR,
+    [
+        ...Object.values(TOKENS)
+    ], "astar")
+
+module.exports = {
+    timetravel: true,
+    methodology: "Versa Tvl Calculation",
+    tvl: astarTvl
```
Comment (g1nt0ki) [inline projects/Versa/index.js:25]: sorry, you missed to fix this line

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
still need to fix module.exports

**Comment by versa-finance-dev** [comment]
Fixed

**Comment by llamatester** [comment]
Error while running adapter at projects/Versa/index.js: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:206:11
        at processTicksAndRejections (node:internal/process/task_queues:96:5)

**Comment by g1nt0ki** [comment]
thanks for the PR, fixed the minor bug after merging

**Pull request merged**